src/WebJobs.Script/Binding/HttpBinding.cs

```diff
@@ -44,51 +48,55 @@ public override Collection<CustomAttributeBuilder> GetCustomAttributes(Type para
             }
 
             HttpStatusCode statusCode = HttpStatusCode.OK;
-            JObject headers = null;
+
             if (content is string)
             {
                 try
                 {
-                    // attempt to read the content as a JObject
-                    JObject jo = JObject.Parse((string)content);
-
-                    // if the content is json we capture that so it will be
-                    // serialized as json by WebApi below
-                    content = jo;
-
-                    // TODO: Improve this logic
-                    // Sniff the object to see if it looks like a response object
-                    // by convention
-                    JToken value = null;
-                    if (jo.TryGetValue("body", StringComparison.OrdinalIgnoreCase, out value))
-                    {
-                        content = value;
+                    // attempt to read the content as a ExpandoObject
+                    content = JsonConvert.DeserializeObject<ExpandoObject>(content as string, new ExpandoObjectConverter());
+                }
+                catch (JsonException)
+                {
+                    // not a json response
+                }
+            }
 
-                        if (value is JValue && ((JValue)value).Type == JTokenType.String)
-                        {
-                            // convert raw strings so they get serialized properly below
-                            content = (string)value;
-                        }
+            if (content is ExpandoObject)
+            {
+                var responseObject = content as ExpandoObject;
```

Comment:
yeah, just looked up casting recommendations, seems like this is the ticket.  Will do throughout